### PR TITLE
Adding ARM storage module and samples

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding ARM storage module, samples and docs.

### CODE REVIEW

File: sdk/arm/storage/2019-06-01/armstorage/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Generated from https://github.com/Azure/azure-rest-api-specs/tree/9563d064dabcf535e35a52bc2e0e3ada4e74e64f
+
+Code generator @autorest/go@4.0.0-preview.4
```
Comment (jhendrixMSFT) [inline sdk/arm/storage/2019-06-01/armstorage/CHANGELOG.md:3]: Regenerate with preview.5?

File: sdk/arm/storage/2019-06-01/armstorage/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## Prerequisites
+
+- an [Azure subscription](https://azure.microsoft.com/free/)
+- Go 1.13 or above
+
+## Provide Feedback
+
+If you encounter bugs or have suggestions, please
+[open an issue](https://github.com/Azure/azure-sdk-for-go/issues) and assign the `Resource Graph` label.
```
Comment (jhendrixMSFT) [inline sdk/arm/storage/2019-06-01/armstorage/README.md:29]: Storage label.

File: sdk/arm/storage/2019-06-01/armstorage/CHANGELOG.md (hunk 1 above)
Comment (catalinaperalta) [reply sdk/arm/storage/2019-06-01/armstorage/CHANGELOG.md]: Done regenerated with preview.6

File: sdk/arm/storage/2019-06-01/armstorage/README.md (hunk 2 above)
Comment (catalinaperalta) [reply sdk/arm/storage/2019-06-01/armstorage/README.md]: Done

File: sdk/arm/storage/2019-06-01/armstorage/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+## Prerequisites
+
+- an [Azure subscription](https://azure.microsoft.com/free/)
+- Go 1.13 or above
+
+## Provide Feedback
+
+If you encounter bugs or have suggestions, please
+[open an issue](https://github.com/Azure/azure-sdk-for-go/issues) and assign the `Storage` label.
+
```
Comment (nickzhums) [inline sdk/arm/storage/2019-06-01/armstorage/README.md:30]: where's the link to the code samples?

File: sdk/arm/storage/2019-06-01/armstorage/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+```sh
+go get -u github.com/Azure/azure-sdk-for-go/sdk/arm/storage/2019-06-01/armstorage
+```
+
+## Prerequisites
+
+- an [Azure subscription](https://azure.microsoft.com/free/)
+- Go 1.13 or above
+
```
Comment (nickzhums) [inline sdk/arm/storage/2019-06-01/armstorage/README.md:25]: we should have a section for authentication, explaining the new Azure Identity support and how we can use that in some code snippets.

File: sdk/arm/storage/2019-06-01/armstorage/README.md (hunk 3 above)
Comment (jhendrixMSFT) [reply sdk/arm/storage/2019-06-01/armstorage/README.md:30]: It's the `PkgGoDev` link at the top.

File: sdk/arm/storage/2019-06-01/armstorage/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+## Provide Feedback
```
Comment (nickzhums) [inline sdk/arm/storage/2019-06-01/armstorage/README.md:26]: We need to have a section for usage pattern (e.g. creating a client and making a call to the service)

File: sdk/arm/storage/2019-06-01/armstorage/README.md (hunk 4 above)
Comment (jhendrixMSFT) [reply sdk/arm/storage/2019-06-01/armstorage/README.md:25]: We can add a link to the documentation for `azidentity`.

File: sdk/arm/storage/2019-06-01/armstorage/README.md (hunk 5 above)
Comment (jhendrixMSFT) [reply sdk/arm/storage/2019-06-01/armstorage/README.md:26]: That's in the samples.  I explicitly didn't want to duplicate it in the README.

File: sdk/arm/storage/2019-06-01/armstorage/README.md (hunk 3 above)
Comment (nickzhums) [reply sdk/arm/storage/2019-06-01/armstorage/README.md:30]: how about making it more obvious ? 

something like 

# Code Samples

For code samples of how to use this package, please see 
[link to code samples]

Comment (jhendrixMSFT) [reply sdk/arm/storage/2019-06-01/armstorage/README.md:30]: Will change this up.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**